### PR TITLE
CotE - Mirumoto Dojo

### LINKED ISSUE #2788: CotE - Mirumoto Dôjô

### PR DESCRIPTION
**Pull request by jamesoddy**
Closes #2788 

Includes PR #2822 (for the DuelTypes enum)

### CODE REVIEW

**Comment by jeremylarner** [comment]
In future, I think it might be better not to mix PRs like this.  I realise not having the previous one causes tests to break, but for review purposes, it makes it really difficult to see what's going on.  I'm happy to be overruled if others disagree though

**Comment by jamesoddy** [comment]
I wass hoping the idea would be to approve and merge in PR 2788 first then this one should just simplify to the implementation and tests for Mirumoto Dojo.  I take your point though - I'll try and avoid it in future.

**Comment by jeremylarner** [comment]
Yeah, I still have 31 files changed :(

**Comment by jamesoddy** [comment]
Yeah, realised it wasn't going to work because the duels PR is going to be squashed when it is merged so the commit history is completely different.  I've sorted it out and now this is just the changes needed for Mirumoto Dojo.

**Comment by jeremylarner** [comment]
Ah, it's my fault for squashing? I'd never made the connection.... Sorry :/

File: server/game/CardAbility.js [hunk 1]
```diff
@@ -9,16 +9,17 @@ class CardAbility extends ThenAbility {
         if(properties.initiateDuel) {
             if(properties.condition) {
                 let condition = properties.condition;
-                properties.condition = context => context.source.isParticipating() && condition(context);
+                properties.condition = context => (context.source.type !== CardTypes.Character || context.source.isParticipating()) && condition(context);
```
Comment (jeremylarner) [inline server/game/CardAbility.js]: Given the target conditions, do we need these extra condition checks? (I know that's not something you added, just pointing it out)

**Comment by jamesoddy** [comment]
Haha, not blaming you for squashing :) .  I think squashing is best to clean up the commits.  I just hadn't considered it in this case that it means that including other PRs in a PR will never clean up once the other PR is merged.

File: server/game/CardAbility.js (hunk 1 above)
Comment (jamesoddy) [reply server/game/CardAbility.js]: Yeah I think you are right.  I'll push a new commit.

**Pull request merged**